### DISCUSSION TITLE
Failed to extract shortcode

### DISCUSSION THREAD

**Discussion by elvisdragonmao** [chipzoller/hugo-clarity#380] [Q&A] [answered]
I have been working on this for hours but I can't solve this problem.
My blog is [here ](https://EM-Tec.github.io)and here's the [repo](https://github.com/EM-Tec/EM-Tec.github.io).

Everyting is working okay for a long time but github action started to fail a week ago. Here's the log of github action.
```
Start building sites … 
hugo v0.101.0-466fa43c16709b4483689930a4f9ac8add[5](https://github.com/EM-Tec/EM-Tec.github.io/actions/runs/3016228630/jobs/4848633963#step:4:6)c9f[6](https://github.com/EM-Tec/EM-Tec.github.io/actions/runs/3016228630/jobs/4848633963#step:4:7)6+extended linux/amd64 BuildDate=2022-06-16T0[7](https://github.com/EM-Tec/EM-Tec.github.io/actions/runs/3016228630/jobs/4848633963#step:4:8):09:16Z VendorInfo=gohugoio
Error: Error building site: "/home/runner/work/EM-Tec.github.io/EM-Tec.github.io/content/about.en.md:12:1": failed to extract shortcode: template for shortcode "notice" not found
Total in 79 ms
Error: Process completed with exit code 255.
```

I try to create a new site at my PC and if I use any shortcode like notice it started to fail again.
here's the log in VSCode
```
PS C:\Users\s01\Documents\GitHub\EM-Tec.github.io> hugo server
Start building sites … 
hugo v0.102.3-b76146b129d7caa52417f8e914fc5b9271bf56fc windows/amd64 BuildDate=2022-09-01T10:16:19Z VendorInfo=gohugoio
Error: Error building site: "C:\Users\s01\Documents\GitHub\EM-Tec.github.io\content\post\remove-bg.md:1:1": execute of template failed: template: _default/_markup/render-image.html:47:4: executing "_default/_markup/render-image.html" at <partial "figure" (dict "file" $file "image" $image "diskPath" $.Page.File.Dir "webPath" $.Page.RelPermalink "alt" $alt "cap" $cap "classes" ($scratch.Get "classes") "bundle" $bundle)>: error calling partial: "C:\Users\s01\Documents\GitHub\EM-Tec.github.io\themes\hugo-clarity\layouts\partials\figure.html:42:15": execute of template failed: template: partials/figure.html:42:15: executing "partials/figure.html" at <fileExists $itemDisk>: error calling fileExists: CreateFile C:\Users\s01\Documents\GitHub\EM-Tec.github.io\post\https:\EM-Tec.github.io\images\remove-bg-api.avif: The filename, directory name, or volume label syntax is incorrect.
Built in 244 ms
```
Very hope that someone can help me >_<

**Comment by rootwork** [comment]
On the action, you have a typo here:

https://github.com/EM-Tec/EM-Tec.github.io/blob/main/.github/workflows/create.yml#L14

`ture` should be `true`

That will ensure the GitHub action grabs the submodules, which is how you're loading the theme. Right now it's not loading the theme, and so the `notice` shortcode is undefined. (I wish Hugo gave more helpful error messages in these situations, like erroring that the theme itself can't be found, but this is where we are.)

On the PC version you have a different error message essentially saying the image you're referencing can't be found. (This is one place where Hugo Clarity could probably do a better version of an error message itself.) I would really triple-check that the path to the image you're referencing is correct, especially paying attention to whether you're using [Page Bundles](https://gohugo.io/content-management/page-bundles/) or not, because that affects how it gets resolved. You might try referencing some other images. If you try multiple images and it's not an issue with using/not using Page Bundles, then it's possible there's a bug in Hugo Clarity's image rendering.

**Reply by elvisdragonmao** [reply]
Thank you for your reply.🙏 Actually, I use back to submodule because of this error.

I don't think my image path have any problem (since there have been no problem for a year). That is a post written long ago. This problem started from [this commit](https://github.com/EM-Tec/EM-Tec.github.io/commit/afbc0e2ec38e3e37edca6d425961eced74ac24b0). I have try adding " because I thought maybe the + confuse it but still not working.

Here's the log after correcting the typle
```
Start building sites … 
hugo v0.101.0-466fa43c16709b4483689930a4f9ac8add5c9f66+extended linux/amd64 BuildDate=2022-06-16T07:09:16Z VendorInfo=gohugoio
Error: Error building site: "/home/runner/work/EM-Tec.github.io/EM-Tec.github.io/content/post/link-to-qr.md:1:1": "/home/runner/work/EM-Tec.github.io/EM-Tec.github.io/themes/hugo-clarity/layouts/_default/_markup/render-image.html:23:20": execute of template failed: template: _default/_markup/render-image.html:23:20: executing "_default/_markup/render-image.html" at <$image.MediaType.MainType>: error calling MediaType: error calling resources.GetRemote: failed to resolve media type for remote resource "https://EM-Tec.github.io/images/link-to-qr-function.png"
Total in 907 ms
```

**Reply by elvisdragonmao** [reply]
@rootwork Is this a bug?

**Reply by rootwork** [reply]
@Edit-Mr Did you update the theme just before or just after that commit, either by redownloading it or updating the git submodule/Hugo module? If not, then I'm not sure how the commit you referenced could have changed the loading of an image elsewhere.

But it's possible the error message could be a red herring. Given that it's just a single tag and it has special characters, try this -- change:

```
tags: C++
```

to:

```
tags: "C++"
```

And see if that changes anything. If you still get the same error message, try this:

```
tags:
  - "C++"
```

All of those *should* work, I'm just trying to eliminate some bug with the tag name/count given that's the commit you say it happened on.

**Reply by elvisdragonmao** [reply]
@rootwork I didn't update anything before commiting.
This is why I'm confused. Commit log is [here](https://github.com/EM-Tec/EM-Tec.github.io/commits/main).

I've try updating hugo and theme to solve this problem. But still....
I don't think my image have any problem. Anything that uses the template code of clarity will cause problem.(Like `{{notice}}` or `![:inline]`)

**Reply by rootwork** [reply]
Well, if you didn't update anything then I'm not sure what to tell you. Theme files wouldn't just change on their own.

Did you try the different methods of listing the tags?

If you're getting errors with _any_ shortcodes, then it's probably something to do with the theme files not being properly there/fully loaded/fully accessible. I would try reinstalling them.

**Comment by elvisdragonmao** [comment, marked as answer]
Ok so for some reason, some of my image was broken. I reupload it and everything is alright.
I rebuild the whole site just in case anything is up to date. Thank you @rootwork
